src/west/commands/project.py

```diff
@@ -1000,11 +1000,14 @@ def _fetch(project, rev=None):
         _update_manifest_rev(project, '{revision}')
     else:
         # The revision is definitely not a SHA, so it's safe to fetch directly.
-        # This avoids fetching unnecessary ref space from the remote.
-        # We need {{commit}} instead of {commit} because everything gets run
-        # through Project.format.
-        fetch_cmd += '{revision}:' + QUAL_MANIFEST_REV
+        # This avoids fetching unnecessary refs from the remote.
+        #
+        # We update manifest-rev to FETCH_HEAD instead of using a
+        # refspec in case the revision is a tag, which we can't use
+        # from a refspec.
+        fetch_cmd += '{revision}'
         project.git(fetch_cmd)
+        _update_manifest_rev(project, 'FETCH_HEAD^{{commit}}')
```

Comment:
You are missing the `We need {{commit}} instead of {commit} because everything gets run through Project.format` comment, which was very informative.

Reply:
> You are missing the We need {{commit}} instead of {commit} because everything gets run through Project.format comment, which was very informative.

@carlescufi fixed, thanks. In the future, I think I want to disable that by default and make this user-selectable with a `fmt=True` keyword argument. It's a bit too magical for an API.

Reply:
> I think I want to disable that by default and make this user-selectable with a `fmt=True` keyword argument. It's a bit too magical for an API.

You know what, on second thought, I want to remove it without making it user selectable. Experimenting with this a bit, we can just use f-strings now that we require Python 3.6 or later.  The results are quite readable and not magic.